pkg/sqlcache/db/client.go

```diff
@@ -353,27 +358,43 @@ func closeRowsOnError(rows Rows, err error) error {
 
 // NewConnection checks for currently existing connection, closes one if it exists, removes any relevant db files, and opens a new connection which subsequently
 // creates new files.
-func (c *client) NewConnection() error {
+func (c *client) NewConnection(useTempDir bool) (string, error) {
 	c.connLock.Lock()
 	defer c.connLock.Unlock()
 	if c.conn != nil {
 		err := c.conn.Close()
 		if err != nil {
-			return err
+			return "", err
 		}
 	}
-	err := os.RemoveAll(InformerObjectCacheDBPath)
-	if err != nil {
-		return err
+	if !useTempDir {
+		err := os.RemoveAll(InformerObjectCacheDBPath)
+		if err != nil {
+			return "", err
+		}
 	}
 
 	// Set the permissions in advance, because we can't control them if
 	// the file is created by a sql.Open call instead.
-	if err := touchFile(InformerObjectCacheDBPath, informerObjectCachePerms); err != nil {
-		return nil
+	var dbPath string
+	if useTempDir {
+		dir := os.TempDir()
+		f, err := os.CreateTemp(dir, InformerObjectCacheDBPathRoot)
```

Comment:
Should we be using `InformerObjectCacheDBPathRoot` if we are just going to be adding `.db` to it later?

Reply:
No, because it provides the prefix for the base temp file:

```
	InformerObjectCacheDBPathRoot := "ioc"
	dir  := os.TempDir()
	f, _ := os.CreateTemp(dir, InformerObjectCacheDBPathRoot)
	fmt.Printf("filename: %s\n", f.Name())
```

=> "/tmp/ioc3178645212" and then we need to append the ".db" extension to keep sqlite happy

Reply:
Also sqlite creates 2 auxiliary files, `informer_object_cache.db-shm` and `informer_object_cache.db-wal` so it looks like filename extensions matter.

Reply:
Sick, works for me :+1: